src/pages/dev/general-message-passing/cosmos-gmp.mdx

```diff
@@ -162,6 +163,40 @@ pub fn multi_send_to_evm(
 ```
 
 ## Chain Support
+
 For a Cosmos blockchain to support receiving GMP messages (naively or via CosmWasm), it should implement the appropriate middleware in Go as part of the consensus layer.
 
-See our [Sample GMP Middleware](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/native-integration/sample-middleware).
\ No newline at end of file
+If you have CosmWasm on your Cosmos chain and want to be able to receieve and deliver messages, install [Osmosis IBC-hooks](https://github.com/osmosis-labs/osmosis/tree/main/x/ibc-hooks).
+
+If you don’t support CosmWasm on your Cosmos chain, you'll need to implement and install your own Axelar GMP module to receive and execute messages. See our [sample reference implementation](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/native-integration/sample-middleware).
+
+## Establishing a Path of Trust
+
+Because IBC messages are permissionless, you'll want to ensure you verify the source of all messages.
+
+When you receive a packet on Cosmos chain, you'll need to verify:
+
+1. That the packet came from Axelar (verify IBC channel; Axelar GMP sender account).
+
+   Specify the Axelar IBC channel by allowlisting the channel hash.
+
+   - **[CosmWasm](https://github.com/osmosis-labs/osmosis/blob/main/x/ibc-hooks/README.md#cosmwasm-contract-execution-format)**
+
+     For example, in your CosmWasm code, you can combine the IBC hook, the channel ID, and the Axelar GMP account to verify the channel matches the expected channel.
+
+     ```rust
+
+     `Bech32(Hash("ibc-wasm-hook-intermediary" || channelID to Axelar || Axelar GMP account))` assuming the channel is fixed, which is usually true for mainnet
+
+     ```
+
+   - **For Consensus**
+
+     Verify (channel & sender == address of our GMP account = `axelar1dv4u5k73pzqrxlzujxg3qp8kvc3pje7jtdvu72npnt5zhq05ejcsn5qme5`.
+     https://github.com/axelarnetwork/evm-cosmos-gmp-sample/blob/main/native-integration/sample-middleware/gmp_middleware.go#L114
+
+1. The EVM source chain and address.
+
+   When a message has an EVM source, the chain and address are passed in differently for CosmWasm and Native GMP.
+   - **CosmWasm**: The CosmWasm contract can define `source_chain` and/or `source_address` as arguments for their `wasm` method. And pass in the correct `source_chain` and/or `source_address` as an argument from the EVM contract. The Axelar network will validate that these values are accurate and not being spoofed by the sender.
```

Comment:
Do we want the developer to pass in the correct `source_chain` as an argument from the EVM contract, or are we saying that the CosmWasm contract will do that? Current wording is not clear.